components/match2/wikis/fifa/match_group_input_custom.lua

```diff
@@ -0,0 +1,624 @@
+		end
+	end
+
+	if record.name and record.type ~= Opponent.literal then
+		record.name = record.name:gsub(' ', '_')
+	end
+
+	if record.type == Opponent.team then
+		record.icon, record.icondark = CustomMatchGroupInput.getIcon(opponent.template)
+		record.match2players = CustomMatchGroupInput._readTeamPlayers(record, record.players)
+	end
+
+	return record
+end
+
+function CustomMatchGroupInput._readTeamPlayers(opponent, playerData)
+	local players = CustomMatchGroupInput._getManuallyEnteredPlayers(playerData)
+
+	if Table.isEmpty(players) then
+		players = CustomMatchGroupInput._getPlayersFromVariables(opponent.name)
+	end
+
+	return players
+end
+
+function CustomMatchGroupInput._getManuallyEnteredPlayers(playerData)
+	local players = {}
+	playerData = Json.parseIfString(playerData) or {}
+
+	for prefix, displayName in Table.iter.pairsByPrefix(playerData, 'p') do
+		local name = mw.ext.TeamLiquidIntegration.resolve_redirect(Logic.emptyOr(
+			playerData[prefix .. 'link'],
+			displayName
+		)):gsub(' ', '_')
+
+		table.insert(players, {
+			name = name,
+			displayname = displayName,
+			flag = Flags.CountryName(playerData[prefix .. 'flag']),
+		})
+	end
+
+	return players
+end
+
+function CustomMatchGroupInput._getPlayersFromVariables(teamName)
+	local teamNameWithSpaces = teamName:gsub('_', ' ')
+	local players = {}
+
+	local playerIndex = 1
+	while true do
+		local prefix = teamName .. '_p' .. playerIndex
+		local prefixWithSpaces = teamNameWithSpaces .. '_p' .. playerIndex
+		local playerName = Variables.varDefault(prefix, Variables.varDefault(prefixWithSpaces))
+		if String.isEmpty(playerName) then
+			break
+		end
+		table.insert(players, {
+			name = playerName:gsub(' ', '_'),
+			displayname = Variables.varDefault(
+					prefix .. 'dn',
+					Variables.varDefault(prefixWithSpaces .. 'dn', playerName:gsub('_', ' '))
+			),
+			flag = Flags.CountryName(Variables.varDefault(prefix .. 'flag', Variables.varDefault(prefixWithSpaces .. 'flag'))),
+		})
+		playerIndex = playerIndex + 1
+	end
+
+	return players
+end
+
+function CustomMatchGroupInput._mapInput(match, mapIndex)
+	local map = Json.parseIfString(match['map' .. mapIndex])
+
+	if Table.isEmpty(map) then
+		match['map' .. mapIndex] = nil
+		return match
+	end
+
+	if Logic.readBool(match.hasSubmatches) then
+		-- generic map name (not displayed)
+		map.map = 'Game ' .. mapIndex
+	elseif Logic.readBool(map.penalty) then
+		map.map = 'Penalties'
+	else
+		map.map = mapIndex .. Ordinal.suffix(mapIndex) .. ' Leg'
+	end
+
+	-- set initial extradata for maps
+	map.extradata = {
+		comment = map.comment,
+		penaltyscores = CustomMatchGroupInput._submatchPenaltyScores(match, map),
+	}
+
+	-- inherit stuff from match data
+	map.type = Logic.emptyOr(map.type, match.type)
+	map.liquipediatier = match.liquipediatier
+	map.liquipediatiertype = match.liquipediatiertype
+	map.game = Logic.emptyOr(map.game, match.game)
+	map.date = Logic.emptyOr(map.date, match.date)
```

Comment:
Use common with parent

Reply:
```suggestion
map = MatchGroupInput.getCommonTournamentVars(map, match)
```

Reply:
added with the `-- inherit stuff from match data` line as description restored as well